### PR TITLE
Feature/sw caching

### PR DESCRIPTION
**Pull request by Matt-Roden**
# Description drawing

- Add in code to successfully cache and view application offline
- User can only see what pokemon they have previously found when offline

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor feature (non-breaking change that modifies existing work)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] local host 3000
- [x] console.log()
- [x] dev tools
- [ ] cypress testing

# Checklist:

- [x] My code follows the style guidelines of this project/linted
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: public/serviceworker.js [hunk 1]
```diff
@@ -1,42 +1,192 @@
...
+];
+
+// on activation we clean up the previously registered service workers
+self.addEventListener('activate', evt =>
+  evt.waitUntil(
+    caches.keys().then(cacheNames => {
+      return Promise.all(
+        cacheNames.map(cacheName => {
+          if (cacheName !== CURRENT_CACHE) {
+            return caches.delete(cacheName);
```
Comment (joanafbrito) [inline public/serviceworker.js:38]: Great super WIN!!!!!

File: public/offline.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="Web site created using create-react-app" />
+    <link rel="apple-touch-icon" type="images/png" href="./Images/favicon.png" />
+    <link rel="manifest" href="./manifest.json" />
+    <title>PokeDex</title>
+  </head>
+  <body>
+    <noscript>You need to enable JavasScript to run this app.</noscript>
+    <h1>Please Pay your dammn bill!</h1>
```
Comment (hayleyw7) [inline public/offline.html:15]: Wait, what is this??? lol

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -29,13 +29,13 @@ const Header = () => {
...
         <img className='ash' src='Images/ash1.png' alt='ash-ketchum'/>
 
-        <img className='pokeball' src='Images/pokeball.png' alt='pokeball'/>    
+        <img className='pokeball' src='Images/pokeball.png' alt='pokeball'/>
 
-      </article>  
+      </article>
 
-      <div class="custom-shape-divider-bottom-1631589819">
+      <div className="custom-shape-divider-bottom-1631589819">
```
Comment (joanafbrito) [inline src/components/Header/Header.js:36]: thanks for correcting that

**Review by joanafbrito — APPROVED** [review]
Awesome job!!!

File: src/components/Header/Header.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ const Header = () => {
       <nav className='nav'>
         <Link
           to={`/help`}
-          key={`help`}  
+          key={`help`}
```
Comment (hayleyw7) [inline src/components/Header/Header.js:16]: Should we change this to `/howto` now?

File: public/serviceworker.js [hunk 5]
```diff
@@ -1,42 +1,192 @@
...
+  '/src/components/PokedexGrid/PokedexGrid.js',
+  '/src/components/PokedexGrid/PokedexGrid.css',
+  '/src/components/PokemonDetails/PokemonDetails.js',
+  '/src/components/PokemonDetails/PokemonDetails.css',
+  '/src/components/PokemonCard/PokemonCard.js',
+  '/src/components/PokemonCard/PokemonCard.css',
+];
+
+// on activation we clean up the previously registered service workers
+self.addEventListener('activate', evt =>
```
Comment (hayleyw7) [inline public/serviceworker.js:32]: eventListener 1

File: public/serviceworker.js [hunk 6]
```diff
@@ -1,42 +1,192 @@
...
         })
-      ))
+      );
+    })
   )
-})
+);
+
+// on install we download the routes we want to cache for offline
+self.addEventListener('install', evt =>
```
Comment (hayleyw7) [inline public/serviceworker.js:47]: eventListener 2

File: public/serviceworker.js [hunk 7]
```diff
@@ -1,42 +1,192 @@
...
+const update = request =>
+  caches
+    .open(CURRENT_CACHE)
+    .then(cache =>
+      fetch(request).then(response => cache.put(request, response))
+    );
+
+// general strategy when making a request (eg if online try to fetch it
+// from the network with a timeout, if something fails serve from cache)
+self.addEventListener('fetch', evt => {
```
Comment (hayleyw7) [inline public/serviceworker.js:86]: eventListener 3

**Review by hayleyw7 — APPROVED** [review]

**Pull request merged**